libs/src/sdk/services/Auth/AppAuth.ts

```diff
@@ -0,0 +1,34 @@
+import { EIP712TypedData, MessageData, signTypedData } from 'eth-sig-util'
+import type { AuthService } from './types'
+
+/**
+ * AuthService for Developer Apps wishing to write on a user's behalf
+ */
+export class AppAuth implements AuthService {
+  private apiKey: string
+  private apiSecret: string
+
+  constructor(apiKey: string, apiSecret: string) {
+    this.apiKey = apiKey
+    this.apiSecret = apiSecret
+  }
+
+  getSharedSecret: (publicKey: string | Uint8Array) => Promise<Uint8Array> =
+    async () => {
+      throw new Error('AppAuth does not support getSharedSecret')
+    }
+
+  sign: (data: string) => Promise<[Uint8Array, number]> = () => {
+    throw new Error('AppAuth does not support sign')
+  }
+
+  signTransaction = async (data: MessageData<EIP712TypedData>['data']) => {
+    return signTypedData(Buffer.from(this.apiSecret, 'hex'), {
+      data
+    })
+  }
+
+  getAddress: () => Promise<string> = async () => {
+    return this.apiKey
```

Comment:
Since `apiKey` is a pubkey, we prob have to convert it to an address right?

Reply:
I wonder if we already have a util for that somewhere, it doesn't look like it. But should be easy enough